crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+module dwallet_system::dwallet {
+    use std::vector;
+    use dwallet::object::{Self, UID, ID};
+    use dwallet::transfer;
+    use dwallet::event;
+    use dwallet::tx_context;
+    use dwallet::tx_context::{TxContext};
+
+    friend dwallet_system::dwallet_2pc_mpc_ecdsa_k1;
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+    const ENotSystemAddress: u64 = 0;
+    const EMesssageApprovalDWalletMismatch: u64 = 1;
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Constants <<<<<<<<<<<<<<<<<<<<<<<<
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Constants <<<<<<<<<<<<<<<<<<<<<<<<
+
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Events <<<<<<<<<<<<<<<<<<<<<<<<
+    struct NewSignSessionEvent<E: store + copy + drop> has copy, drop {
+        session_id: ID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        messages: vector<vector<u8>>,
+        sender: address,
+        sign_data_event: E,
+    }
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Events <<<<<<<<<<<<<<<<<<<<<<<<
+
+    struct DWalletCap has key, store {
+        id: UID,
+    }
+
+    struct ApprovalsHolder has key {
+        id: UID,
+        message_approvals: vector<MessageApproval>,
+    }
+
+    struct MessageApproval has store {
+        dwallet_cap_id: ID,
+        message: vector<u8>,
+    }
+
+    struct SignMessages<S: store, E: store + copy + drop> has key, store {
```

Comment:
Why is this object named liked this? its a very confusing name. @omersadika 
We should rename. Perhaps to `MessagesForSigning` or `ApprovedMessagesToSign` or something like that.